test/end2end_test.go

```diff
@@ -7433,29 +7445,30 @@ func (s *httpServer) start(t *testing.T, lis net.Listener) {
 	}()
 }
 
-func doHTTPHeaderTest(t *testing.T, errCode codes.Code, headerFields ...[]string) {
+func doHTTPHeaderTest(t *testing.T, errCode codes.Code, headerFields ...[]string) error {
```

Comment:
Instead of changing this function to return an error and calling t.Error() from the test, does it make sense to annotate this function with a call to t.Helper(), and thereby the error output will contain the line number of the caller instead of line numbers from here.

https://golang.org/pkg/testing/#T.Helper

Reply:
This is nice. I didn't know this. Done!